server/webhook_parser.go

```diff
@@ -289,21 +294,35 @@ func appendCommentNotifications(wh *webhook, verb string) {
 		wh.notifications = append(wh.notifications, notification)
 	}
 
-	// Don't send a notification to the assignee if they don't exist, or if are also the author.
-	// Also, if the assignee was mentioned above, avoid sending a duplicate notification here.
+	commentMessage := fmt.Sprintf("%s **commented** on %s:\n>%s", commentAuthor, jwh.mdKeySummaryLink(), jwh.Comment.Body)
+
+	// Don't send a notification to the assignee or reporter if they don't exist, or if they are also the author.
+	// Also, if the assignee or reporter was mentioned above, avoid sending a duplicate notification here.
 	// Jira Server uses name field, Jira Cloud uses the AccountID field.
-	if assigneeMentioned || jwh.Issue.Fields.Assignee == nil || jwh.Issue.Fields.Assignee.Name == jwh.User.Name ||
-		(jwh.Issue.Fields.Assignee.AccountID != "" && jwh.Issue.Fields.Assignee.AccountID == jwh.Comment.UpdateAuthor.AccountID) {
-		return
+	if !assigneeMentioned && shouldSendNotification(jwh.Issue.Fields.Assignee, jwh.User.Name, jwh.Comment.UpdateAuthor.AccountID) {
+		wh.notifications = append(wh.notifications, webhookUserNotification{
+			jiraUsername:  jwh.Issue.Fields.Assignee.Name,
+			jiraAccountID: jwh.Issue.Fields.Assignee.AccountID,
+			message:       commentMessage,
+			postType:      PostTypeComment,
+			commentSelf:   jwh.Comment.Self,
+			recipientType: recipientTypeAssignee,
+		})
+	}
+	if !reporterMentioned && shouldSendNotification(jwh.Issue.Fields.Reporter, jwh.User.Name, jwh.Comment.UpdateAuthor.AccountID) {
+		wh.notifications = append(wh.notifications, webhookUserNotification{
+			jiraUsername:  jwh.Issue.Fields.Reporter.Name,
+			jiraAccountID: jwh.Issue.Fields.Reporter.AccountID,
+			message:       commentMessage,
+			postType:      PostTypeComment,
+			commentSelf:   jwh.Comment.Self,
+			recipientType: recipientTypeReporter,
+		})
 	}
+}
 
-	wh.notifications = append(wh.notifications, webhookUserNotification{
-		jiraUsername:  jwh.Issue.Fields.Assignee.Name,
-		jiraAccountID: jwh.Issue.Fields.Assignee.AccountID,
-		message:       fmt.Sprintf("%s **commented** on %s:\n>%s", commentAuthor, jwh.mdKeySummaryLink(), jwh.Comment.Body),
-		postType:      PostTypeComment,
-		commentSelf:   jwh.Comment.Self,
-	})
+func shouldSendNotification(to *jira.User, fromName, fromAccountID string) bool {
+	return !(to == nil || to.Name == fromName || (to.AccountID != "" && to.AccountID == fromAccountID))
```

Comment:
To fix the issue with Jira cloud, we need to add an extra check. I suggest making these a little more verbose since we made it its own function:

```suggestion
    if to == nil {
        return false
    }
    
    // check for Jira Server user
    if to.Name != "" && to.Name == fromName {
        return false
    }    
    
    // check for Jira Cloud user
    if to.AccountID != "" && to.AccountID == fromAccountID {
        return false
    }
    
    return true
}
```

The check for `to.Name != ""` is the actual fix here. With this fix, assignees and reporters should get DM notifications on Jira Cloud. cc @DHaussermann